### PR TITLE
[Improvement] improvement in PARTITION ALL

### PR DESCRIPTION
**Pull request by awakeljw**
### Changelog category (leave one):

- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
ASTPartition::formatImpl should output ALL while executing ALTER TABLE t DETACH PARTITION ALL

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by kssenii** [comment]
@Mergifyio update

**Comment by awakeljw** [comment]
@tavplubix I have fix it according to your suggestion.

File: src/Storages/MergeTree/MergeTreeData.cpp [hunk 1]
```diff
@@ -3385,22 +3385,30 @@ void MergeTreeData::checkAlterPartitionIsPossible(
             else
             {
                 /// We are able to parse it
-                getPartitionIDFromQuery(command.partition, getContext());
+                const auto * partition_ast = command.partition->as<ASTPartition>();
+                if (partition_ast &&  partition_ast->all)
```
Comment (kssenii) [inline src/Storages/MergeTree/MergeTreeData.cpp:3389]:
```suggestion
                if (partition_ast && partition_ast->all)
```

**Review by kssenii — COMMENTED** [review]
Let's try to remove these lines https://github.com/awakeljw/ClickHouse/blob/9a4cc78dfb083a20c7f8ffa9cea5baaeca958ba5/src/Parsers/ParserPartition.cpp#L35-L38 too. If there is ALL, they should not be present.

File: src/Storages/MergeTree/MergeTreeData.cpp (hunk 1 above)
Comment (awakeljw) [reply src/Storages/MergeTree/MergeTreeData.cpp]: I agree with your suggestion and have fix it.

**Review by kssenii — APPROVED** [review]

**Comment by kssenii** [comment]
tests failures not related (broken in master)

**Pull request merged**